src/EditorFeatures/CSharpTest/ImplementAbstractClass/ImplementAbstractClassTests_ThroughMember.cs

```diff
@@ -0,0 +1,670 @@
+                 SingleOption(CSharpCodeStyleOptions.PreferExpressionBodiedProperties, CSharpCodeStyleOptions.NeverWithSilentEnforcement),
+                 SingleOption(CSharpCodeStyleOptions.PreferExpressionBodiedIndexers, CSharpCodeStyleOptions.NeverWithSilentEnforcement));
+
+        internal Task TestAllOptionsOffAsync(
+            string initialMarkup,
+            string expectedMarkup,
+            IDictionary<OptionKey, object> options = null,
+            ParseOptions parseOptions = null)
+        {
+            options ??= new Dictionary<OptionKey, object>();
+            foreach (var kvp in AllOptionsOff)
+            {
+                options.Add(kvp);
+            }
+
+            return TestInRegularAndScriptAsync(
+                initialMarkup,
+                expectedMarkup,
+                index: 1,
+                options: options,
+                parseOptions: parseOptions);
+        }
+
+        [Fact, WorkItem(41420, "https://github.com/dotnet/roslyn/issues/41420")]
+        public async Task FieldInBaseClassIsNotSuggested()
+        {
+            await TestExactActionSetOfferedAsync(
+@"abstract class Base
+{
+    public Base Inner;
+
+    public abstract void Method();
+}
+
+class [|Derived|] : Base
+{
+}", new[] { FeaturesResources.Implement_abstract_class });
+        }
+
+        [Fact, WorkItem(41420, "https://github.com/dotnet/roslyn/issues/41420")]
+        public async Task FieldInMiddleClassIsNotSuggested()
+        {
+            await TestExactActionSetOfferedAsync(
+@"abstract class Base
+{
+    public abstract void Method();
+}
+
+abstract class Middle : Base
+{
+    public Base Inner;
+}
+
+class [|Derived|] : Base
+{
+}", new[] { FeaturesResources.Implement_abstract_class });
+        }
+
+        [Fact, WorkItem(41420, "https://github.com/dotnet/roslyn/issues/41420")]
+        public async Task FieldOfSameDerivedTypeIsSuggested()
+        {
+            await TestInRegularAndScriptAsync(
+@"abstract class Base
+{
+    public abstract void Method();
+}
+
+class [|Derived|] : Base
+{
+    Derived inner;
+}",
+@"abstract class Base
+{
+    public abstract void Method();
+}
+
+class Derived : Base
+{
+    Derived inner;
+
+    public override void Method()
+    {
+        inner.Method();
+    }
+}", index: 1, title: string.Format(FeaturesResources.Implement_through_0, "inner"));
+        }
+
+        [Fact, WorkItem(41420, "https://github.com/dotnet/roslyn/issues/41420")]
+        public async Task SkipInaccessibleMember()
+        {
+            await TestInRegularAndScriptAsync(
+@"abstract class Base
+{
+    public abstract void Method1();
+    protected abstract void Method2();
+}
+
+class [|Derived|] : Base
+{
+    Base inner;
+}",
+@"abstract class Base
+{
+    public abstract void Method1();
+    protected abstract void Method2();
+}
+
+class {|Conflict:Derived|} : Base
```

Comment:
Should we add another test that this code fix is not offered on the resulting code, i.e. all members are inaccessible and hence would be skipped?

Reply:
sure.